yorkie/src/test/kotlin/dev/yorkie/document/json/JsonTreeTest.kt

```diff
@@ -300,6 +316,142 @@ class JsonTreeTest {
         )
     }
 
+    @Test
+    fun `should emit Tree edit operations accordingly when edited with index`() = runTest {
+        val document = Document(Document.Key(""))
+        fun JsonObject.tree() = getAs<JsonTree>("t")
+
+        document.updateAsync {
+            it.setNewTree(
+                "t",
+                ElementNode(
+                    "doc",
+                    children = listOf(
+                        ElementNode(
+                            "p",
+                            children = listOf(TextNode("ab")),
+                        ),
+                    ),
+                ),
+            )
+        }.await()
+        assertEquals("<doc><p>ab</p></doc>", document.getRoot().tree().toXml())
+
+        val actualOperationDeferred = async(UnconfinedTestDispatcher()) {
+            document.events("$.t")
+                .filterIsInstance<LocalChange>()
+                .flatMapConcat { it.changeInfo.operations.asFlow() }
+                .filter { it is TreeEditOpInfo || it is TreeStyleOpInfo }
+                .take(2)
+                .toList()
+        }
+
+        document.updateAsync {
+            it.tree().edit(1, 1, TextNode("X"))
+            assertEquals("<doc><p>Xab</p></doc>", it.tree().toXml())
+
+            it.tree().style(0, 0, mapOf("a" to "b"))
+        }.await()
+
+        assertContentEquals(
+            listOf(
+                TreeEditOpInfo(
+                    1,
+                    1,
+                    listOf(0, 0),
+                    listOf(0, 0),
+                    TreeNode("text", value = "X"),
+                    "$.t",
+                ),
+                TreeStyleOpInfo(
+                    0,
+                    0,
+                    listOf(0, -7), // TODO: need to check if these are the actual right values???
```

Comment:
@7hong13  we need to check whether `fromPath`, `toPath`  returned with events are actual values when editing with index.

Reply:
I'm not sure it can have negative values. I'll look up to it later.

Reply:
@skhugh 
I think it results in negative numbers since `from` and `to` indexes are invalid.
the range should be on the closing tag, like `it.tree().style(4, 5, mapOf("a" to "b"))` in this case.
then the `OpInfo` should be like below:
```kotlin
TreeStyleOpInfo(
         4,
         5,
         listOf(0),
         listOf(0),
         mapOf("a" to "b"),
         "$.t",
),
```

I examined the JS SDK, and it also yields negative value paths for the same case.